Shipturfs and various patchie

Latentish opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Makes ship specific turfs so we don't have /plating/ planetary turfs with infinite atmos.
Replaces planetary and plating turfs on the Glaive and Shepherd.
Swaps the captain's spare the Montagnes starts with for a gold ID.
Fixes a runtime caused by attacking water with anything that isn't a container.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
We didn't have any proper planetary type turfs that could be used on ships. This caused infinite atmos issues and a bandaid patch that left "plating" and "wetplating" that don't have smoothing instead. Also minor patches for the ships this touches, and a fix to a water tile related runtime are good for the game.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl:phoaly
add: Added shipturfs
del: Old planetary turf subtypes
fix: Fixed water turfs runtiming when hit with a bat
fix: Replaced tiles on the Glaive and Shepherd
tweak: Swapped the montagnes ID
tweak: Nerfed Glaive mining gear
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Latentish committed:
Shiptiles!

-Adds various ship friendly versions of turfs
- replaces actuallydark versions of turfs with ones that don't have infinite atmos.

Latentish committed:
Merge remote-tracking branch 'upstream/master' into Glaive-Patchie

Latentish committed:
Revert "Merge remote-tracking branch 'upstream/master' into Glaive-Patchie"

This reverts commit 06c698c36a19e47568f71ecd39179cfee56241a3, reversing
changes made to 883c34a246ba83932414c64a919cf1cff73d8769.

Latentish committed:
Revert "Revert "Merge remote-tracking branch 'upstream/master' into Glaive-Patchie""

This reverts commit 65746f725229e4d3bc13ccaae6f75ea06504cf0c.

Latentish committed:
Various fixes

- Makes grass and dirt shipturfs /floor/ship instead of /plating/ship.
-Ship dirt cannot be dug up but can be crowbarred out for regular plating.
-Grass can be dug up for shipdirt
-Fixes a runtime with water tiles
-Makes a proper /ship/water
-removes superfluous code

Latentish committed:
Requested changes

-Adds a second entrance.
-Adds a shuttle docking port.

Latentish commented:
I will NOT atomize my changes.

Zevotech commented:
![image](https://github.com/shiptest-ss13/Shiptest/assets/95449138/4c4f5e83-1271-4862-9ff2-8fe37ae5b556)
gender????

Latentish committed:
Updates dirt behavior

-Makes shovels useable with shipdirt

Erikafox reviewed (approved):
map changes lgtm

Latentish committed:
Merge remote-tracking branch 'upstream/master' into Glaive-Patchie

Latentish committed:
Re-Does shep stuff

-Re-replaces shep tiles.

Latentish committed:
removes a couple redundant lines of code

-Titular

Latentish committed:
Merge branch 'master' into Glaive-Patchie

tmtmtl30 commented on code/game/turfs/open/floor/misc_floor.dm:
```diff
@@ -190,3 +190,122 @@
+/turf/open/floor/ship/dirt/dark
+	icon_state = "greenerdirt"
+
+/turf/open/floor/grass/ship
```
is this intended to be /turf/open/floor/ship/grass?

Latentish replied:
No. Greener dirt is a subtype of jungle dirt.  This is how it is supposed to be.

tmtmtl30 commented on code/game/turfs/open/floor/misc_floor.dm:
```diff
@@ -190,3 +190,122 @@
+	var/smooth_icon = 'icons/turf/floors/grass.dmi'
+	baseturfs = /turf/open/floor/ship/dirt
+
+/turf/open/floor/grass/ship/Initialize(mapload, inherited_virtual_z)
```
same thing with the type. also, curious how these are removed?

Latentish replied:
These are removed with a shovel, also they are separate from the normal /grass because those do not smooth, don't have  a shipturf dirt baseturf, and result in a shipturf dirt with shoveled.

Latentish replied:
![image](https://github.com/shiptest-ss13/Shiptest/assets/55075747/1357f7b3-128b-4e98-8614-25a60a29c6ce)

tmtmtl30 commented on code/game/turfs/open/floor/misc_floor.dm:
```diff
@@ -190,3 +190,122 @@
+	baseturfs = /turf/open/floor/ship/dirt/dark
+	smooth_icon = 'icons/turf/floors/junglegrass.dmi'
+
+/turf/open/floor/plating/ship/water
```
why is this a /plating, unlike the others, out of curiosity?

Latentish replied:
The water is a plating because unlike the others it's a plating filled with water. It can be built upon to remove. It also makes it so that it cannot have wires built through it.

tmtmtl30 reviewed (commented):
not too familiar with all the stuff turfs should have, but there's nothing egregious here (other than a possible typing error?)

Latentish committed:
patchie again

-Fixes missing tiles on shepherd
-Replaces mini pickaxes for normal ones
-Replaces advanced auto mining scanners for regular auto mining scanners
-replaces heavy mining hardsuit with independent mining hardsuit
-replaces bounty hunting masks with regular breath masks
-Removes 1 intelligence potion

spockye committed:
Merge branch 'master' into Glaive-Patchie

tmtmtl30 commented:
looks good to me, probably
